addons/registry/enable

```diff
@@ -1,48 +1,79 @@
-		read -r key value <<<$(echo $arg | gawk -F "=" '{print $1 ,$2}')
-    if [ "$key" != "size" ]; then
-      echo "The only authorized argument is \"size\", eg  microk8s.enable registry:size=30Gi"
-      echo "The \"size\" value should match the regex: ([a-zA-Z]+=[a-zA-Z0-9]+(,|))"
-      exit 1
-    elif [ "$key" = "size"  ] && [[ ! $value =~ $REGEX_DISK_SIZE ]]; then
-      echo "The size of the registry should be higher or equal to 20Gi and match the regex: (^[2-9][0-9]{1,}|^[1-9][0-9]{2,})(Gi$)"
-      exit 1
-		elif [ "$key" = "size" ] && [[ $value =~ $REGEX_DISK_SIZE ]]; then
-      "$SNAP/microk8s-enable.wrapper" hostpath-storage
-      declare -A map
-      map[\$DISKSIZE]=$value
-    fi
-  done
-  echo "Enabling the private registry"
-  echo "Applying registry manifest"
-  use_addon_manifest registry/registry apply "$(declare -p map)"
-  echo "The registry is enabled"
-  echo "The size of the persistent volume is $value"
-else
-  echo "The only authorized argument is \"size\", eg  microk8s.enable registry:size=30Gi"
-  echo "The \"size\" value should match the regex: ([a-zA-Z]+=[a-zA-Z0-9]+(,|))"
-  exit 1
-fi
+#!/usr/bin/env python3
+import os
+import subprocess
+import shlex
+import re
+import yaml
+
+import click
+
+REGEX_DISK_SIZE = "(^[2-9][0-9]{1,}|^[1-9][0-9]{2,})(Gi$)"
+
+
+@click.command()
+@click.option("--size", default="20Gi", help="Size of the registry")
+@click.option("--storageclass", default=None, help="Storage class to be used")
+def enable_registry(size, storageclass):
+    snap_path = os.environ.get("SNAP")
+    snap_data = os.environ.get("SNAP_DATA")
+    snap_user_data = os.environ.get("SNAP_USER_DATA")
+    current_path = os.path.dirname(os.path.realpath(__file__))
+    manifest = "{}/registry.yaml".format(current_path)
+
+    # Enable hostpath-storage first. If already enabled, nothing changes.
+    subprocess.call(
+        ["{}/microk8s-enable.wrapper".format(snap_path), "hostpath-storage"]
+    )
+
+    command = shlex.split(
+        '{}/microk8s-kubectl.wrapper --kubeconfig={}/credentials/client.config get sc --no-headers -o custom-columns=":metadata.name"'.format(
+            snap_path, snap_data
+        )
+    )
+    sclasses = subprocess.check_output(command)
+    sclasses = sclasses.decode("utf-8")
+    sclasses = sclasses.split("\n")
+    sclasses = [sc.strip() for sc in sclasses if sc.strip()]
+
+    if (storageclass is not None) & (storageclass not in sclasses):
+        print("Error: Storage class {} doesn't exit.".format(storageclass))
+        return
+
+    if not bool(re.match(REGEX_DISK_SIZE, size)):
+        print(
+            "The size of the registry should be higher or equal to 20Gi and match the regex: {}".format(
+                REGEX_DISK_SIZE
+            )
+        )
+        return
+
+    with open(manifest, "r", encoding="utf8") as f:
+        docs = list(yaml.safe_load_all(f))
+        docs[1].get("spec").get("resources").get("requests")["storage"] = size
```

Comment:
I see the reason behind doing this, but it does not really save us from anything. If `docs[1]["spec"]` is empty, then the next chained `.get()` will fail. Let's not pretend to avoid any of the issues here and instead just do dict lookups for everything.